src/sentry/static/sentry/app/components/group/externalIssueActions.tsx

```diff
@@ -37,87 +37,146 @@ class ExternalIssueActions extends AsyncComponent<Props, State> {
+      selectedIntegration: null,
     });
   };
 
   handleClick = (action: 'create' | 'link') => {
     this.setState({action});
   };
 
-  renderBody() {
-    const {action, selectedIntegration, issue} = this.state;
+  linkIssueSuccess = (
+    integration: GroupIntegration,
+    externalIssue: IntegrationExternalIssue
+  ) => {
+    let linked = JSON.parse(JSON.stringify(integration)) as GroupIntegration;
+    linked.externalIssues = [externalIssue];
+    this.setState(
+      {
+        linked: [...this.state.linked, linked],
+        unlinked: this.state.unlinked.filter(config => config.id !== linked.id),
+      },
+      () => this.closeModal()
+    );
+  };
 
+  renderBody() {
+    const {action, selectedIntegration, linked, unlinked} = this.state;
     return (
       <React.Fragment>
-        <IssueSyncListElement
-          onOpen={this.openModal}
-          externalIssueLink={issue ? issue.url : null}
-          externalIssueId={issue ? issue.id : null}
-          externalIssueKey={issue ? issue.key : null}
-          externalIssueDisplayName={issue ? issue.displayName : null}
-          onClose={this.deleteIssue.bind(this)}
-          integrationType={selectedIntegration.provider.key}
-          hoverCardHeader={t('Linked %s Integration', selectedIntegration.provider.name)}
-          hoverCardBody={
-            issue && issue.title ? (
-              <div>
-                <IssueTitle>{issue.title}</IssueTitle>
-                {issue.description && (
-                  <IssueDescription>{issue.description}</IssueDescription>
-                )}
-              </div>
-            ) : (
-              <IntegrationItem integration={selectedIntegration} />
-            )
-          }
-        />
+        {linked.length > 0 &&
+          linked.map(config => {
+            const {provider, externalIssues} = config;
+            let issue = externalIssues[0];
+            return (
+              <IssueSyncListElement
+                key={issue.id}
+                externalIssueLink={issue.url}
+                externalIssueId={issue.id}
+                externalIssueKey={issue.key}
+                externalIssueDisplayName={issue.displayName}
+                onClose={() => this.deleteIssue(config)}
+                integrationType={provider.key}
+                hoverCardHeader={t('Linked %s Integration', provider.name)}
+                hoverCardBody={
+                  <div>
+                    <IssueTitle>{issue.title}</IssueTitle>
+                    {issue.description && (
+                      <IssueDescription>{issue.description}</IssueDescription>
+                    )}
+                  </div>
+                }
+              />
+            );
+          })}
+
+        {unlinked.length > 0 && (
+          <IssueSyncListElement
+            externalIssueLink={null}
+            externalIssueId={null}
+            externalIssueKey={null}
+            externalIssueDisplayName={null}
```

Comment:
@NisanthanNanthakumar it would be better if `IssueSyncListElement` accepted `undefined` instead of `null` here so we could omit passing values to these props bt I'll leave that as an optional suggestion.

Reply:
@scefali we use `IssueSyncListElement` elsewhere and they explicitly write `value` or `null`. So I think I'll keep the `null` for consistency.

Reply:
@NisanthanNanthakumar I just realized that `externalIssueId`, `externalIssueKey`, and `externalIssueDisplayName` are optional props and we don't actually set the prop values in `src/sentry/static/sentry/app/components/group/externalIssuesList.tsx`.